java/src/test/java/io/cucumber/java/JavaDefaultDataTableEntryTransformerDefinitionTest.java

```diff
@@ -131,6 +136,33 @@ void method_must_have_cell_transformer_as_optional_third_argument() throws Throw
         assertThrows(InvalidMethodSignatureException.class, () -> new JavaDefaultDataTableEntryTransformerDefinition(method, lookup));
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {"testString", "TestString", "Test String", "test String", "Test string"})
+    void convert_to_camel_case(String header) throws Throwable {
+        String expectedHeader = "testString";
+        String actualHeader = toCamelCase(header);
+        Assert.assertThat(actualHeader, equalTo(expectedHeader));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+        "threeWordsString", "ThreeWordsString", "three Words String", "Three Words String", "Three words String",
+        "Three Words string", "Three words string", "three Words string", "three words String", "threeWords string",
+        "three WordsString", "three wordsString",
+    })
+    void convert_three_words_to_camel_case(String header) throws Throwable {
+        String expectedHeader = "threeWordsString";
+        String actualHeader = toCamelCase(header);
+        Assert.assertThat(actualHeader, equalTo(expectedHeader));
+    }
+
+    private String toCamelCase(String header) throws Throwable{
+        Constructor constructor = JavaDefaultDataTableEntryTransformerDefinition.class.getDeclaredClasses()[0].getDeclaredConstructor();
+        constructor.setAccessible(true);
+        Object camelCaseStringConverter = constructor.newInstance();
+        Method method = camelCaseStringConverter.getClass().getDeclaredMethod("toCamelCase", String.class);
+        return (String) method.invoke(camelCaseStringConverter, header);
```

Comment:
I don't think is necessary to use violence to test the `CamelCaseStringConverter`.

You may want to consider moving `convertToCamelCase` into the `CamelCaseStringConverter` and making the class a package private member of the `io.cucumber.java` package. That way you can focus on testing the title case to camel case edge cases withouth having to construct the `JavaDefaultDataTableEntryTransformerDefinition.`